[verydaftpunk] iP

verydaftpunk opened the pull request.

Commit:
Add Gradle support

verydaftpunk committed:
level 1: greet, echo, exit

verydaftpunk committed:
assume no more than 100 tasks

verydaftpunk committed:
change bot name

verydaftpunk committed:
Add, List, Mark as done (yet to do unmark)

verydaftpunk committed:
new class called Task

verydaftpunk committed:
fixed small error - print to println for completeTask() method

verydaftpunk committed:
added a method called unstrike in my Task class to basically unmark ("unstrike") a task

verydaftpunk committed:
implementing unstrike (basically "unmark") method in Task class

verydaftpunk committed:
added a class called ToDo which is a child of Task

verydaftpunk committed:
a new class called Deadline which is a child of Task

verydaftpunk committed:
added a class called Event that is a child of Task

verydaftpunk committed:
added a method to Task to get the [ ] or [X] sign from a given task

verydaftpunk committed:
added custom methods for adding a to do, adding a deadline, adding an event

verydaftpunk committed:
edited constructor for Event to take in start time and end time

verydaftpunk committed:
edited the toString method for Task class because there was a slight error with spacing earlier

verydaftpunk committed:
fixed a spacing error for the methods named completetask and undocompletetask

verydaftpunk committed:
added the text "here are the tasks in your list: " when displaying the list to user

verydaftpunk committed:
changed the updating of numOfThings (number of things) in list to fix a small error in numbering items when displaying list to user

verydaftpunk committed:
fixed the issue of numOfThings basically never updating (always zero)

verydaftpunk committed:
level 5 - updated the message if there is nothing in the list

verydaftpunk committed:
level 5 - updated the message if user marks a task that does not exist

verydaftpunk committed:
added a try catch if user does not type a todo

verydaftpunk committed:
level 5 - added try catch if user fails to input deadline/event

verydaftpunk committed:
changed the array of tasks to arraylist. this makes it easier to delete and add tasks. added a method called delete as well

verydaftpunk committed:
added a method called set rank to Task class to update rank of a task especially after delete is used

verydaftpunk committed:
setRank method in Task class to update the rank of a Task

verydaftpunk committed:
level 7

verydaftpunk committed:
level 8

verydaftpunk committed:
Merge pull request #1 from verydaftpunk/branch-Level-8

level 8

verydaftpunk committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  level 8

verydaftpunk committed:
typo

verydaftpunk committed:
typo

verydaftpunk committed:
package skittles

verydaftpunk committed:
no message

verydaftpunk committed:
Gradle

verydaftpunk committed:
add test files

verydaftpunk committed:
javadoc

verydaftpunk committed:
javadoc

verydaftpunk committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  javadoc

verydaftpunk committed:
javadoc

verydaftpunk committed:
Merge commit '556af3f47a96b32898ab4cdbd65b16486a4871e8' into branch-Level-9

* commit '556af3f47a96b32898ab4cdbd65b16486a4871e8':
  Add Gradle support

verydaftpunk committed:
level 9

verydaftpunk committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  level 9
  javadoc

Diatbbin commented on src/main/java/skittles/Duke.java:
```diff
@@ -0,0 +1,303 @@
+                    //show user the list in this case
+                    ListOfStuff.displayLst();
+                    break;
+                case MARK:
```
Dont need to indent for case clauses. Need to add a space after switch.

Diatbbin commented on src/main/java/skittles/Instruction.java:
```diff
@@ -0,0 +1,49 @@
+    DELETE,
+    BYE,
+    FIND;
+
```
Good use of enums!

Diatbbin commented on src/test/java/skittles/ToDoTest.java:
```diff
@@ -0,0 +1,15 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+public class ToDoTest {
+    @Test
+    public void dummyTest(){
+        assertEquals(2, 2);
+    }
```
Rmb to add actual test cases!!

Diatbbin commented on src/main/java/skittles/Duke.java:
```diff
@@ -0,0 +1,303 @@
+public class Duke {
+    public Duke() {}
+
+    private static final Scanner takingInput = new Scanner(System.in);
```
Constant names must be all uppercase using underscore to separate words.

Diatbbin reviewed (commented):
Overall, code LGTM! Code looks clean and neat. Great use of OOP! Do rmb to remove code that has been commented out!

verydaftpunk committed:
DeadlineTest and ToDoTest

verydaftpunk committed:
Level 10

verydaftpunk committed:
level 10

verydaftpunk committed:
level 10

YZmunchmunch commented on src/main/java/Deadline.java:
```diff
@@ -1,14 +1,64 @@
+     * @param dateline String of date of deadline given by user.
+     * @return A Date object with the same details as what the user has given.
+     */
+    public Date convertStringIntoDateFormat(String dateline) {
```
I get it that you are trying to convert the string input of the date into a Date object, then converting it back into a string format for printing, perhaps these 2 methods can be combined, which can make your code neater since lesser methods are being called as well.

YZmunchmunch commented on src/main/java/Deadline.java:
```diff
@@ -1,14 +1,64 @@
+     */
     @Override
     public String convertToText() {
         return String.format("D | %d | %s | %s", super.getDoneOrNot() ? 1 : 0, super.getName(),
```
Perhaps the getDoneOrNot() method could be renamed to simply be getDone()!

YZmunchmunch reviewed (commented):
Overall, this is pretty neat code, even though certain methods could be made simpler by joining. However, structure of the code seems good!

verydaftpunk committed:
Currently there are two dependencies as I simply copy and pasted the part to add previously without incorporating it into the current dependencies

There should be only one dependencies even though having two separate ones work, because it is clearer to any person reading the code

I have combined both dependencies into one

This makes the code more reader friendly and organised

verydaftpunk committed:
now when gui runs, it greets

verydaftpunk committed:
week 5

verydaftpunk committed:
Sort

verydaftpunk committed:
Add files via upload

verydaftpunk committed:
Update README.md
